tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -586,7 +516,8 @@ Status StorageManager::array_xlock(const URI& array_uri) {
   // Wait until the array is closed
   std::unique_lock<std::mutex> lk(open_array_mtx_);
   xlock_cv_.wait(lk, [this, array_uri] {
-    return open_arrays_.find(array_uri.to_string()) == open_arrays_.end();
+    return open_arrays_for_reads_.find(array_uri.to_string()) ==
```

Comment:
Does the exclusive lock also need to wait for the array to not be in the `open_arrays_for_writes_` map?

Reply:
No it doesn't. The reason is that the xlock is acquired so that old fragments (and hence also their metadata) are not deleted at the final step of consolidation. These metadata are only held by opened arrays in read mode, so there is no reason to wait on arrays opened in write mode. I will add a comment.